test/integration/acceptance/add_signatory_test.cpp

```diff
@@ -0,0 +1,172 @@
+  const std::string kUser2 = "usertwo";
+  const std::string kUser2Id = kUser2 + "@test";
+  const std::string kAmount = "1.0";
+  const crypto::Keypair kUser2Keypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+#define CHECK_BLOCK(i) \
+  [](auto &block) { ASSERT_EQ(block->transactions().size(), i); }
+
+/**
+ * C224 Add existing public key of other user
+ * @given some user with CanAddSignatory permission and a second user
+ * @when execute tx with AddSignatory where the first is a creator and the
+ *       second's key is added as a signatory
+ * @then there is the tx in proposal
+ */
+TEST_F(AddSignatory, Basic) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(makeFirstUser(), CHECK_BLOCK(1))
+      .sendTxAwait(makeSecondUser(), CHECK_BLOCK(1))
+      .sendTxAwait(
+          complete(baseTx().addSignatory(kUserId, kUser2Keypair.publicKey()),
+                   kUserKeypair),
+          CHECK_BLOCK(1));
+}
+
+/**
+ * C228 AddSignatory without such permissions
+ * @given some user without CanAddSignatory permission and a second user
+ * @when execute tx with AddSignatory where the first is a creator and the
+ *       second's key is added as a signatory
+ * @then there is the no tx in proposal
+ */
+TEST_F(AddSignatory, NoPermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(makeFirstUser({interface::permissions::Role::kReceive}),
+                   CHECK_BLOCK(1))
+      .sendTxAwait(makeSecondUser(), CHECK_BLOCK(1))
+      .sendTx(
+          complete(baseTx().addSignatory(kUserId, kUser2Keypair.publicKey()),
+                   kUserKeypair))
+      .checkVerifiedProposal(CHECK_BLOCK(0));
+}
+
+/**
+ * C225 Add signatory to other user
+ * C227 Add signatory to an account, which granted permission to add it, and add
+ *      the same public key
+ * @given some user with CanAddMySignatory permission and a second user with
+          granted CanAddMySignatory
+ * @when execute tx with AddSignatory where the second is a creator and the
+ *       second's key is added as a signatory
+ * @then there is the tx in proposal
+ */
+TEST_F(AddSignatory, GrantedPermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(
+          makeFirstUser({interface::permissions::Role::kAddMySignatory}),
+          CHECK_BLOCK(1))
+      .sendTxAwait(makeSecondUser(), CHECK_BLOCK(1))
+      .sendTxAwait(
+          complete(baseTx().grantPermission(
+              kUser2Id, interface::permissions::Grantable::kAddMySignatory)),
+          CHECK_BLOCK(1))
+      .sendTxAwait(complete(baseTx().creatorAccountId(kUser2Id).addSignatory(
+                                kUserId, kUser2Keypair.publicKey()),
+                            kUser2Keypair),
+                   CHECK_BLOCK(1));
+}
+
+/**
+ * C226 Add signatory to account, which haven't granted such permission
```

Comment:
"did not grant" is more correct from here imo

Reply:
Passive voice is required, also I would even change it to `isn't granted`